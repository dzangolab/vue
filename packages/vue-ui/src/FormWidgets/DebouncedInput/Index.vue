<template>
  <div :aria-label="ariaLabel ?? placeholder" class="debounce-input">
    <input
      :disabled="disabled"
      :placeholder="placeholder"
      :type="type"
      :value="modelValue"
      @input="onInput"
    />
  </div>
</template>

<script lang="ts">
export default {
  name: "DebouncedInput",
};
</script>

<script setup lang="ts">
import { useDebouncedValue } from "../../utils";

import type { PropType } from "vue";

const props = defineProps({
  ariaLabel: {
    default: "input",
    type: String,
  },
  debounceTime: {
    default: 500,
    required: false,
    type: Number,
  },
  disabled: {
    default: false,
    type: Boolean,
  },
  modelValue: {
    default: "",
    required: false,
    type: [String, Number] as PropType<string | number | null | undefined>,
  },
  placeholder: {
    default: "Search...",
    required: false,
    type: String,
  },
  type: {
    default: "text",
    required: false,
    type: String,
  },
});

const emit = defineEmits(["update:modelValue"]);

const onInput = (event: Event) => {
  const value = (event.target as HTMLInputElement).value;

  useDebouncedValue(value, props.debounceTime).then((debouncedValue) => {
    if (debouncedValue.value) {
      emit("update:modelValue", value);
    }
  });
};
</script>
