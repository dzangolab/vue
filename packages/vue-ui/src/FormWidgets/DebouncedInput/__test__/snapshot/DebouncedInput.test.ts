import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import DebouncedInput from "../../Index.vue";

describe("DebouncedInput", () => {
  it("matches snapshot", () => {
    const wrapper = mount(DebouncedInput);

    expect(wrapper.element).toMatchSnapshot();
  });
});
