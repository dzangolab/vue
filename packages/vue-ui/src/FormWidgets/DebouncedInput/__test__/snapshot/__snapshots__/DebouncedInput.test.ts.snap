// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DebouncedInput > matches snapshot 1`] = `
<div
  aria-label="input"
  class="debounce-input"
>
  <input
    placeholder="Search..."
    type="text"
  />
</div>
`;
